tests/suites/test_suite_cipher.function

```diff
@@ -1100,9 +1104,9 @@ void test_vec_crypt( int cipher_id, int operation, data_t *key,
 
 exit:
     mbedtls_cipher_free( &ctx );
-#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#if defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_DEPRECATED_REMOVED)
```

Comment:
Shouldn't we have the same guard here as for the call to `psa_crypt_init()`, that is `USE_PSA_CRYPTO && !DEPRECATED_REMOVED && TEST_DEPRECATED`? (Or just `USE_PSA_CRYPTO && TEST_DEPRECATED`?)